dirty_cat/super_vectorizer.py

```diff
@@ -257,8 +275,20 @@ def fit_transform(self, X, y=None):
             usually because transformers passed do not match any column.
             To fix the issue, try passing the least amount of None as encoders.
         """
-        X = self._transform(X)
+        # Create a copy to avoid altering the original data.
+        X = X.copy()
```

Comment:
Copies are expensive, we try to do as little as possible (but still, copy when necessary). Maybe it would be possible to move this copy in an "else" block, right below, as I below that the "if" statement is inducing a copy.